core/backend/src/internal/workspace/WorkspaceImpl.ts

```diff
@@ -558,12 +559,15 @@ class EditorContainerImpl extends WorkspaceContainerImpl implements EditableWork
   public override getWorkspaceDb(props: WorkspaceDbProps): EditableWorkspaceDb {
     return this.getEditableDb(props);
   }
+
   public getEditableDb(props: WorkspaceDbProps): EditableWorkspaceDb {
     const db = this._wsDbs.get(workspaceDbNameWithDefault(props.dbName)) as EditableDbImpl | undefined ?? new EditableDbImpl(props, this);
     const isPrerelease = semver.major(db.version) === 0 || semver.prerelease(db.version);
 
-    if (!isPrerelease && this.cloudContainer && this.cloudContainer.queryDatabase(db.dbFileName)?.state !== "copied")
+    if (!isPrerelease && this.cloudContainer && this.cloudContainer.queryDatabase(db.dbFileName)?.state !== "copied") {
+      this._wsDbs.clear();
```

Comment:
why do you clear this?

Reply:
https://github.com/iTwin/itwinjs-core/pull/7004#discussion_r1702066904 if i dont clear, the order of the tests i added mattered

Reply:
```ts
      this._wsDbs.delete(props.dbName);
```

Reply:
yeah, right. only get rid of the problematic db.